torchaudio/functional.py

```diff
@@ -558,37 +557,37 @@ def lfilter(waveform, a_coeffs, b_coeffs):
 
     device = waveform.device
     dtype = waveform.dtype
-    n_channels, n_frames = waveform.size()
+    n_channel, n_sample = waveform.size()
     n_order = a_coeffs.size(0)
     assert(n_order > 0)
 
     # Pad the input and create output
-    padded_waveform = torch.zeros(n_channels, n_frames + n_order - 1, dtype=dtype, device=device)
+    padded_waveform = torch.zeros(n_channel, n_sample + n_order - 1, dtype=dtype, device=device)
     padded_waveform[:, (n_order - 1):] = waveform
-    padded_output_waveform = torch.zeros(n_channels, n_frames + n_order - 1, dtype=dtype, device=device)
+    padded_output_waveform = torch.zeros(n_channel, n_sample + n_order - 1, dtype=dtype, device=device)
 
     # Set up the coefficients matrix
     # Flip order, repeat, and transpose
-    a_coeffs_filled = a_coeffs.flip(0).repeat(n_channels, 1).t()
-    b_coeffs_filled = b_coeffs.flip(0).repeat(n_channels, 1).t()
+    a_coeffs_filled = a_coeffs.flip(0).repeat(n_channel, 1).t()
+    b_coeffs_filled = b_coeffs.flip(0).repeat(n_channel, 1).t()
 
     # Set up a few other utilities
-    a0_repeated = torch.ones(n_channels, dtype=dtype, device=device) * a_coeffs[0]
-    ones = torch.ones(n_channels, n_frames, dtype=dtype, device=device)
+    a0_repeated = torch.ones(n_channel, dtype=dtype, device=device) * a_coeffs[0]
+    ones = torch.ones(n_channel, n_sample, dtype=dtype, device=device)
```

Comment:
I saw `frame`, `time`, `sample` are used. Do they refer to the same thing?

Reply:
Before this PR, `time` was used to indicate time dimension. `n_frames` and `n_samples` were used inconsistently to refer to the number of time sample available. Here, I'm suggesting to use `frame` (and `n_frame`) to refer to "groups of samples" (e.g. in STFT, we have rolling windows of samples to operate on), and "n_sample" to the number of samples in a "time" dimension.

We could also decide to change "time" dimension to "sample" dimension, but that does not seem to have been a source of confusion, so I would avoid changing this in this PR.